services/QuillLMS/app/workers/sync_school_data_from_url_worker.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class SyncSchoolDataFromUrlWorker
+  include Sidekiq::Worker
+
+  def perform(url)
+    response = HTTParty.get(url).parsed_response
+
+    if response['next']
+      SyncSchoolDataFromUrlWorker.perform_async(response['next'])
+    end
+
+    schools_data = response['results']
+
+    schools_data.each do |school_data|
+      attributes_hash = build_attribute_hash(school_data)
+
+      district = District.find_by(nces_id: school_data['leaid'].to_i)
+      attributes_hash[:district_id] = district&.id
+
+      # we have to do this complicated search because historically nces ids, which are largely but not always numerals stored as strings in our database, have sometimes been stored with leading zeroes and sometimes without
```

Comment:
You could extract a constant out for the SQL query:
`NCES_ID_BLAH_QUERY = "CAST(CASE WHEN nces_id ~ '^[0-9]+$' THEN nces_id ELSE '0' END AS BIGINT) = ?"`

Then some refactoring here: (I think `find_by` should work here in place of `.where(...).first`)
```
nces_id = attributes_hash[:nces_id]
school = School.find_by(nces_id:) || School.find_by(NCES_ID_BLAH_QUERY, nces_id) 
school.present? ? school.update!(attributes_hash.except(:name)) : School.create!(attributes_hash)
```